cdc/log.cc

```diff
@@ -1316,20 +1325,25 @@ cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout,
                 auto& s = m.schema();
                 tracing::trace(tr_state, "CDC: Generating log mutations for {}", m.decorated_key());
                 int generated_count;
+                stats::part_type_set touched_parts;
                 if (should_split(m, *s)) {
                     tracing::trace(tr_state, "CDC: Splitting {}", m.decorated_key());
                     generated_count = 0;
                     for_each_change(m, s, [&] (mutation mm, api::timestamp_type ts, bytes tuuid, int& batch_no) {
-                        mutations.push_back(trans.transform(std::move(mm), rs.get(), ts, tuuid, batch_no));
+                        auto [mut, parts] = trans.transform(std::move(mm), rs.get(), ts, tuuid, batch_no);
+                        mutations.push_back(std::move(mut));
                         ++generated_count;
+                        touched_parts.add(parts);
                     });
                 } else {
                     tracing::trace(tr_state, "CDC: No need to split {}", m.decorated_key());
                     int batch_no = 0;
                     auto ts = find_timestamp(*s, m);
                     auto tuuid = timeuuid_type->decompose(generate_timeuuid(ts));
-                    mutations.push_back(trans.transform(m, rs.get(), ts, tuuid, batch_no));
+                    auto [mut, parts] = trans.transform(m, rs.get(), ts, tuuid, batch_no);
+                    mutations.push_back(std::move(mut));
                     generated_count = 1;
+                    touched_parts = parts;
```

Comment:
touched_parts = std::move(parts) ?

Reply:
It should not matter here, `touched_parts` is of `enum_set` type, which is essentially a bit-set that uses a single `size_t` under the hood. It is very cheap to copy.

Reply:
OK. Makes sense.